var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -195,7 +182,7 @@ def determine_bootstrap_options(self, spec, withLibs, options):
         options.append("--with-libraries=%s" % ','.join(withLibs))
 
         if '+python' in spec:
-            options.append('--with-python=%s' % python_exe)
+            options.append('--with-python=%s' % spec['python'].executable)
```

Comment:
Looks the problem also exists for Spack-built Python. Apparently `spec['python']` and `spec['python'].package` are very different. I think I have a fix. I'll push it out after some more testing.